hls4ml/templates/vivado/nnet_utils/nnet_graph.h

```diff
@@ -0,0 +1,444 @@
+    #pragma HLS ARRAY_PARTITION variable=data1_logits complete dim=0
+    nnet::dense_resource<data_T, data_T, typename CONFIG_T::dense_config2>(data0, data1_logits, weights1, biases1);
+    data_T data1[CONFIG_T::dense_config2::n_out];
+    #pragma HLS ARRAY_PARTITION variable=data1 complete dim=0
+    nnet::relu<data_T, data_T, typename CONFIG_T::relu_config2>(data1_logits, data1);
+
+    data_T data2_logits[CONFIG_T::dense_config3::n_out];
+    #pragma HLS ARRAY_PARTITION variable=data2_logits complete dim=0
+    nnet::dense_resource<data_T, data_T, typename CONFIG_T::dense_config3>(data1, data2_logits, weights2, biases2);
+    data_T data2[CONFIG_T::dense_config3::n_out];
+    #pragma HLS ARRAY_PARTITION variable=data2 complete dim=0
+    nnet::relu<data_T, data_T, typename CONFIG_T::relu_config3>(data2_logits, data2);
+
+    nnet::dense_resource<data_T, res_T, typename CONFIG_T::dense_config4>(data2, res, weights3, biases3);
+  }
+
+  template<class data_T, class index_T, class res_T, typename CONFIG_T>
+    void edge_aggregate(
+            data_T    edge_attr_1D[CONFIG_T::n_edge*CONFIG_T::edge_dim],
+            index_T   edge_index_1D[CONFIG_T::n_edge*2],
+            res_T     edge_attr_aggr_1D[CONFIG_T::n_node*CONFIG_T::edge_dim])
+  {
+    //initialize arrays
+    // 1. edge_attr (input)
+    data_T edge_attr[CONFIG_T::n_edge][CONFIG_T::edge_dim];
+    #pragma HLS ARRAY_PARTITION variable=edge_attr complete dim=0
+    nnet::vec_to_mat<data_T, data_T, typename CONFIG_T::edge_attr_config>(edge_attr_1D, edge_attr);
+
+    //2. num_edge_per_node (intermediate), 3. edge_aggr_mask (intermediate)
+    index_T num_edge_per_node[CONFIG_T::n_node];
+    #pragma HLS ARRAY_PARTITION variable=num_edge_per_node complete dim=0
+    ap_uint<1> edge_aggr_mask[CONFIG_T::n_node];
+    #pragma HLS ARRAY_PARTITION variable=edge_aggr_mask complete dim=0
+    for(int i=0; i<CONFIG_T::n_node; i++){
+      #pragma HLS UNROLL
+      num_edge_per_node[i] = 0;
+      if(CONFIG_T::aggr==aggr_max){
+        edge_aggr_mask[i] = 0;
+      }
+    }
+
+    //4. edge_attr_aggr (output)
+    res_T edge_attr_aggr[CONFIG_T::n_node][CONFIG_T::edge_dim];
+    #pragma HLS ARRAY_PARTITION variable=edge_attr_aggr complete dim=0
+    if((CONFIG_T::aggr==aggr_sum)||(CONFIG_T::aggr==aggr_mean)){
+      for(int i=0; i < CONFIG_T::n_node; i++){
+        for(int j=0; j<CONFIG_T::edge_dim; j++){
+          #pragma HLS UNROLL
+          edge_attr_aggr[i][j] = 0;
+        }
+      }
+    }
+    else{ //CONFIG_T:aggr==aggr_max, we want to initialize this with the most negative number we can represent
+      // note: works for ap_fixed or float
+      res_T most_negative_num;
+
+      std::ostringstream mnm_T_ss;
+      mnm_T_ss << typeid(most_negative_num).name();
+      if (mnm_T_ss.str()=="f"){
+        most_negative_num = -FLT_MAX+1;
+      }
+      else{
+        most_negative_num.V = 1;
+        most_negative_num.V <<= res_T::width - 1;
+      }

```

Comment:
I think you can replace this with a line like this:

https://github.com/fastmachinelearning/hls4ml/blob/6ea1a1d898a5933f6fea0f55184ef86d5e4cf6bd/hls4ml/templates/vivado/nnet_utils/nnet_pooling_stream.h#L508

Namely, 
```c++
res_T most_negative_num = hls::numeric_limits<typename CONFIG_T::accum_t>::min(); 
```